classes/PaymentModule.php

```diff
@@ -1085,9 +1085,16 @@ protected function createOrderCartRules(
             //  This is an "amount" reduction, not a reduction in % or a gift
             // THEN
             //  The voucher is cloned with a new value corresponding to the remainder
-            $remainingValue = $cartRule->reduction_amount - $values[$cartRule->reduction_tax ? 'tax_incl' : 'tax_excl'];
+            $cartRuleReductionAmountConverted = $cartRule->reduction_amount;
+            if ((int) $cartRule->reduction_currency !== $cart->id_currency) {
```

Comment:
I don't think it's correct as It's the same as a bug you fixed recently The cart currency and/or the cart rule currency can be different from the default one So you need to convert the cart rule price to the default currency, then to the cart currency

Actually as this is not the first time we encounter this kind of behaviour I think we should add a method `Tools::convertPriceToCurrency($price, Currency $fromCurrency, Currency $toCurrency)` (potentially allow `$toCurrency` to be null with fallback to default)
The funny thing is that there is a deprecated note about this `convertPriceToCurrency` method since 1.7.4 but I can't find it anywhere

Reply:
There is a `Tools::convertPriceFull($amount, Currency $currency_from = null, Currency $currency_to = null)` I will use that one ;)

Reply:
done

Reply:
Nice!!
Can you add a `use` statement please? To avoid the `\Currency` notation